torch/onnx/_internal/exporter.py

```diff
@@ -468,14 +585,70 @@ def export(self) -> ExportOutput:
         )
 
         # Export TorchScript graph to ONNX ModelProto.
-        onnx_model = onnxscript_graph.to_model_proto(self.options.opset_version)
+        onnx_model = onnxscript_graph.to_model_proto(
+            self.options.opset_version,
+            include_initializers=self.options.fake_context is None,
+        )
+
         return torch.onnx.ExportOutput(
             onnx_model,
             self.options.fx_tracer.input_adapter,
             self.options.fx_tracer.output_adapter,
             self.options.diagnostic_context,
         )
 
+    def _assert_fake_tensor_mode(self):
+        has_any_fake_tensor = any(
+            isinstance(arg, torch._subclasses.FakeTensor)
+            for arg in self.model_args
+            if arg is not None
+        )
+        has_any_fake_tensor = has_any_fake_tensor or any(
+            isinstance(val, torch._subclasses.FakeTensor)
+            for val in self.model_kwargs.values()
+            if val is not None
+        )
+        if isinstance(self.model, torch.nn.Module):
+            has_any_fake_param = any(
+                isinstance(param, torch._subclasses.FakeTensor)
+                for param in self.model.parameters()  # type: ignore[union-attr]
+                if param is not None
+            )
+        else:
+            has_any_fake_param = False
+        if (
+            has_any_fake_tensor or has_any_fake_param
+        ) and not self.options.fake_context:
+            return OnnxExporterError(
+                self.options.diagnostic_context,
+                "Cannot export a model with fake inputs/weights without enabling fake mode.",
+            )
+        has_any_non_fake_tensors = any(
+            not isinstance(arg, torch._subclasses.FakeTensor)
```

Comment:
similar comment to use pytree

might want to change lambda to `isinstance(arg, torch.Tensor) and not isinstance(arg, FakeTensor)` to exclude false alarming for python scalars, strs, etc.